src/libsyntax/parse/parser.rs

```diff
@@ -2686,27 +2693,31 @@ impl<'a> Parser<'a> {
         let ex = match self.token {
             token::Not => {
                 self.bump();
+                let (interpolated, prev_span) = (self.token.is_interpolated(), self.span);
```

Comment:
I am a little worried about whether this pattern is fragile -- i.e. if future PR authors may not know when to use a similar pattern when adding code to the parser. 

I'm on a mobile device right now, but later today I will try to look over the rest of the parser code and brainstorm whether there might be a way to modify the API so that people won't overlook the choice of span to use